app/models/physical_server.rb

```diff
@@ -27,6 +27,9 @@ class PhysicalServer < ApplicationRecord
 
   scope :with_hosts, -> { where("physical_servers.id in (select hosts.physical_server_id from hosts)") }
 
+  virtual_column :v_availability, :type => :string
```

Comment:
You should add `:uses => :host` to both of these lines to force eager loading. This'll prevent n+1 queries when the widgets are generated.